modules/transport-netty4/src/internalClusterTest/java/org/elasticsearch/http/netty4/Netty4ChunkedContinuationsIT.java

```diff
@@ -315,14 +317,16 @@ public void onFailure(Exception exception) {
 
     private static Releasable withResourceTracker() {
         assertNull(refs);
+        final ReachabilityChecker reachabilityChecker = new ReachabilityChecker();
         final var latch = new CountDownLatch(1);
-        refs = AbstractRefCounted.of(latch::countDown);
+        refs = LeakTracker.wrap(reachabilityChecker.register(AbstractRefCounted.of(latch::countDown)));
         return () -> {
             refs.decRef();
             try {
                 safeAwait(latch);
             } finally {
                 refs = null;
+                reachabilityChecker.ensureUnreachable();
```

Comment:
Nit: I wonder whether we should run the check only when safeAwait fails to avoids potential failure from `ensureUnreachable` itself? So something like
```suggestion
            boolean success = false;
            try {
                safeAwait(latch);
                success = true;
            } finally {
                refs = null;
                if (success == false) {
                    reachabilityChecker.ensureUnreachable();
                }
```
?

Reply:
Yes good call